src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -323,11 +323,20 @@ internal void Update(double number)
                             }
                         }
 
-                        lock (this.histogramBuckets.LockObject)
+                        var sw = default(SpinWait);
+                        while (true)
                         {
-                            this.runningValue.AsLong++;
-                            this.histogramBuckets.RunningSum += number;
-                            this.histogramBuckets.RunningBucketCounts[i]++;
+                            if (Interlocked.Exchange(ref this.histogramBuckets.UsingHistogram, 1) == 0)
+                            {
+                                this.runningValue.AsLong++;
+                                this.histogramBuckets.RunningSum += number;
+                                this.histogramBuckets.RunningBucketCounts[i]++;
+
+                                this.histogramBuckets.UsingHistogram = 0;
+                                break;
+                            }
+
+                            sw.SpinOnce();
```

Comment:
we "spin", but don't use SpinWait in existing code path : https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/MetricPoint.cs#L293-L298

I think the right is to use the SpinWait, as it is more smart.

Reply:
Yeah, we could update it for the other lock-free updates too.

Reply:
thanks. Please create an issue for tracking.

Reply:
Created #2958 to track this issue.